### ISSUE TITLE
Why version 1.2 is not visible in my pg14 ?

### ISSUE THREAD

**Issue by albert-kam** [xocolatl/periods#26]
This is out of curiosity, but after installing postgresql-14-periods in debian buster, the available versions do not include 1.2, only 1.0 and 1.1. Is this intentional ?

Below are the details of my system, package files, and extension infos.

```
Linux localdebian 4.19.0-18-amd64 #1 SMP Debian 4.19.208-1 (2021-09-29) x86_64 GNU/Linux
Debian 10 Buster
```

```
# dpkg -L postgresql-14-periods
/.
/usr
/usr/lib
/usr/lib/postgresql
/usr/lib/postgresql/14
/usr/lib/postgresql/14/lib
/usr/lib/postgresql/14/lib/bitcode
/usr/lib/postgresql/14/lib/bitcode/periods
/usr/lib/postgresql/14/lib/bitcode/periods/periods.bc
/usr/lib/postgresql/14/lib/bitcode/periods.index.bc
/usr/lib/postgresql/14/lib/periods.so
/usr/share
/usr/share/doc
/usr/share/doc/postgresql-14-periods
/usr/share/doc/postgresql-14-periods/README.md.gz
/usr/share/doc/postgresql-14-periods/changelog.Debian.amd64.gz
/usr/share/doc/postgresql-14-periods/changelog.Debian.gz
/usr/share/doc/postgresql-14-periods/changelog.gz
/usr/share/doc/postgresql-14-periods/copyright
/usr/share/doc/postgresql-doc-14
/usr/share/doc/postgresql-doc-14/extension
/usr/share/doc/postgresql-doc-14/extension/README.periods.gz
/usr/share/postgresql
/usr/share/postgresql/14
/usr/share/postgresql/14/extension
/usr/share/postgresql/14/extension/periods--1.0--1.1.sql
/usr/share/postgresql/14/extension/periods--1.0.sql
/usr/share/postgresql/14/extension/periods--1.1--1.2.sql
/usr/share/postgresql/14/extension/periods--1.1.sql
/usr/share/postgresql/14/extension/periods--1.2.sql
/usr/share/postgresql/14/extension/periods.control
/usr/share/doc/postgresql-14-periods/README.periods.gz
```

```
# select * from pg_extension where extname = 'periods';
   oid   | extname | extowner | extnamespace | extrelocatable | extversion |                     extconfig                     |    extcondition     
---------+---------+----------+--------------+----------------+------------+---------------------------------------------------+---------------------
 2406036 | periods |       10 |         2200 | f              | 1.1        | {2406063,2406069,2406089,2406101,2406114,2406137} | {"","","","","",""}
(1 row)
```

```
# alter extension periods update;
NOTICE:  version "1.1" of extension "periods" is already installed
ALTER EXTENSION
```

```
# SELECT * FROM pg_available_extension_versions WHERE name ='periods';
  name   | version | installed | superuser | relocatable | schema |   requires   |                               comment                                
---------+---------+-----------+-----------+-------------+--------+--------------+----------------------------------------------------------------------
 periods | 1.0     | f         | t         | f           |        | {btree_gist} | Provide Standard SQL functionality for PERIODs and SYSTEM VERSIONING
 periods | 1.1     | t         | t         | f           |        | {btree_gist} | Provide Standard SQL functionality for PERIODs and SYSTEM VERSIONING
(2 rows)
```

**Comment by df7cb** [comment]
Hi, could you share the output of `apt-cache policy postgresql-14-periods` ?

**Comment by albert-kam** [comment]
Hi, sure thing !
```

apt-cache policy postgresql-14-periods
postgresql-14-periods:
  Installed: 1.2-3.pgdg100+2+b1
  Candidate: 1.2-3.pgdg100+2+b1
  Version table:
 *** 1.2-3.pgdg100+2+b1 500
        500 http://apt.postgresql.org/pub/repos/apt buster-pgdg/main amd64 Packages
        100 /var/lib/dpkg/status

```

**Comment by df7cb** [comment]
That's the correct version, and works for me here.

Does /usr/share/postgresql/14/extension/periods.control say `default_version = 1.2`?

Does `select pg_read_file('/usr/share/postgresql/14/extension/periods.control');` give the same info?

Is that really the local PostgreSQL database, and not running on some remote host or inside some container where some older package is installed?

**Comment by albert-kam** [comment]
Hi again,

```
cat /usr/share/postgresql/14/extension/periods.control 
comment = 'Provide Standard SQL functionality for PERIODs and SYSTEM VERSIONING'
default_version = 1.2
module_pathname = '$libdir/periods'
relocatable = false
requires = 'btree_gist'
```

```
psql -d mydb -p 5434 -c "select pg_read_file('/usr/share/postgresql/14/extension/periods.control')"
                                   pg_read_file                                   
----------------------------------------------------------------------------------
 comment = 'Provide Standard SQL functionality for PERIODs and SYSTEM VERSIONING'+
 default_version = 1.2                                                           +
 module_pathname = '$libdir/periods'                                             +
 relocatable = false                                                             +
 requires = 'btree_gist'                                                         +
 
(1 row)
```
The cluster is running on my localhost as I do not use containers at the moment, as can be seen here:

```
# pg_lsclusters 
Ver Cluster Port Status Owner    Data directory              Log file
11  main    5433 online postgres /var/lib/postgresql/11/main /var/log/postgresql/postgresql-11-main.log
12  main    5434 online postgres /var/lib/postgresql/12/main /var/log/postgresql/postgresql-12-main.log
14  main    5432 online postgres /var/lib/postgresql/14/main /var/log/postgresql/postgresql-14-main.log
```

I thought maybe I have older version of periods, so I tried these:

```
apt-get update
Hit:1 http://deb.debian.org/debian buster InRelease
Hit:2 http://dl.google.com/linux/chrome/deb stable InRelease                                                                                                                   
Get:3 http://deb.debian.org/debian-security buster/updates InRelease [65.4 kB]                                                                                                 
Get:4 http://repo.mysql.com/apt/debian buster InRelease [22.1 kB]                                                                                                              
Hit:5 https://download.docker.com/linux/debian buster InRelease                                                                                                                
Get:6 http://deb.debian.org/debian buster-updates InRelease [51.9 kB]                                                                                                          
Get:7 http://packages.microsoft.com/repos/vscode stable InRelease [3,959 B]                                       
Err:4 http://repo.mysql.com/apt/debian buster InRelease                                                                         
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 467B942D3A79BD29
Hit:8 http://apt.postgresql.org/pub/repos/apt buster-pgdg InRelease
Get:9 http://packages.microsoft.com/repos/vscode stable/main amd64 Packages [291 kB]
Hit:10 https://repo.windscribe.com/debian buster InRelease
Fetched 434 kB in 1s (342 kB/s)
Reading package lists... Done
```

```
apt --only-upgrade install postgresql-14-periods
Reading package lists... Done
Building dependency tree       
Reading state information... Done
postgresql-14-periods is already the newest version (1.2-3.pgdg100+2+b1).
0 upgraded, 0 newly installed, 0 to remove and 18 not upgraded.
```

**Comment by albert-kam** [comment]
I dont get it, after dropdb, createdb and restore, the 1.2 is now available again on my local cluster ..
